monai/utils/safeeval.py

```
@@ -101,6 +102,7 @@ def safe_eval(
 
     if rewrite_np:
         parsed = _RewriteConstNp(int_type_str, float_type_str).visit(parsed)
+        ast.fix_missing_locations(parsed)
         locals_vars = {"np": np, **(locals_vars or {})}
 
-    return eval(expr, dict(globals_vars) if globals_vars else None, locals_vars)
+    return eval(compile(parsed, "<safe_eval>", "eval"), dict(globals_vars) if globals_vars else None, locals_vars)
```

Ruff (0.16.1)

[error] Use of possibly insecure function; consider using `ast.literal_eval`

(S307)